### ISSUE TITLE
java.lang.IllegalArgumentException: argument type mismatch received in JMX Bean test in Hotspot, but none in OpenJ9

### ISSUE THREAD

**Issue by Mesbah-Alam** [eclipse/openj9#1566]
**Overview**
A JLM test, in which a call to `javax.management.MBeanServerConnection.invoke(..)` is made, ends up producing an `javax.management.RuntimeMBeanException: java.lang.IllegalArgumentException: argument type mismatch` error when it is run with OpenJDK8 Hotspot SDK (and also OpenJDK9 Hotspot SDK), but works fine when run with OpenJDK8-OpenJ9 (and also with OpenJDK9-OpenJ9) SDK.

**Job output with failure (using OpenJDK8 hotspot vm):** 
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/446/console

**Job output without failure (using OpenJDK8 OpenJ9 vm):** 
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/448/console

**Failure stack trace**
```
javax.management.RuntimeMBeanException: java.lang.IllegalArgumentException: argument type mismatch
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrow(DefaultMBeanServerInterceptor.java:839)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrowMaybeMBeanException(DefaultMBeanServerInterceptor.java:852)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:821)
        at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:801)
        at net.adoptopenjdk.test.jlm.resources.ThreadData.invokeMBeanOperation(ThreadData.java:1071)
        at net.adoptopenjdk.test.jlm.resources.ThreadData.writeMBeanInfo(ThreadData.java:886)
        at net.adoptopenjdk.test.jlm.resources.ThreadData.writeData(ThreadData.java:427)
        at net.adoptopenjdk.test.jlm.local.VMLogger.writeServerData(VMLogger.java:236)
        at net.adoptopenjdk.test.jlm.local.VMLogger.main(VMLogger.java:130)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71)
        at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275)
        at com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:193)
        at com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:175)
        at com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:117)
        at com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:54)
        at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
        at com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
        at com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
        at javax.management.StandardMBean.invoke(StandardMBean.java:405)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:819)
        ... 6 more
Exception in thread "main" java.lang.AssertionError: java.lang.IllegalArgumentException: argument type mismatch
        at org.junit.Assert.fail(Assert.java:88)
        at net.adoptopenjdk.test.jlm.resources.ThreadData.invokeMBeanOperation(ThreadData.java:1099)
        at net.adoptopenjdk.test.jlm.resources.ThreadData.writeMBeanInfo(ThreadData.java:886)
        at net.adoptopenjdk.test.jlm.resources.ThreadData.writeData(ThreadData.java:427)
        at net.adoptopenjdk.test.jlm.local.VMLogger.writeServerData(VMLogger.java:236)
        at net.adoptopenjdk.test.jlm.local.VMLogger.main(VMLogger.java:130)
```

**Java versions** 

Fails with OpenJDK8 Hotspot : 

```
openjdk version "1.8.0-adoptopenjdk"
OpenJDK Runtime Environment (build 1.8.0-adoptopenjdk-jenkins_2018_03_08_17_02-b00)
OpenJDK 64-Bit Server VM (build 25.71-b00, mixed mode)

```

Fails with OpenJDK9 Hotspot:
```
openjdk version "9-adoptopenjdk"
OpenJDK Runtime Environment (build 9-adoptopenjdk+0-adhoc.jenkins.openjdk)
OpenJDK 64-Bit Server VM (build 9-adoptopenjdk+0-adhoc.jenkins.openjdk, mixed mode)
```

Passes with OpenJDK8 OpenJ9 : 

```
openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-b12)
Eclipse OpenJ9 VM (build master-9c05da60, JRE 1.8.0 Linux amd64-64 Compressed References 20180326_129 (JIT enabled, AOT enabled)
OpenJ9   - 9c05da60
OMR      - 03270bf3
JCL      - 8599c6f6 based on jdk8u162-b12)

```

Passes with OpenJDK9 OpenJ9 : 

```
openjdk version "9.0.4-adoptopenjdk"
OpenJDK Runtime Environment (build 9.0.4-adoptopenjdk+0-adhoc.jenkins.openjdk)
Eclipse OpenJ9 VM (build master-9979700b, JRE 9 Linux amd64-64 Compressed References 20180328_147 (JIT enabled, AOT enabled)
OpenJ9   - 9979700b
OMR      - 30c4f205
JCL      - f4b533d1c based on jdk-9.0.4+12)
```

**To reproduce**

You can reproduce the defect by running the personal build at Adopt (https://ci.adoptopenjdk.net/view/work%20in%20progress/job/test_personal/).

To run with OpenJDK 8 Hotspot SDK, please use the following values:

```
personalRepo = Mesbah-Alam
personalBranch = addJlmTestToAdoptOpenJDK
PLATFORM = x64_linux
SPEC = linux_x86-64_cmprssptrs
JAVA_VERSION = SE80
JAVA_IMPL = hotspot
JVM_VERSION = openjdk8
TESTPROJECT = systemtest 
TARGET = TestJlmLocal_0
MACHINE_LABEL  = linux&&x64&&test
SDK_RESOURCE = nightly 
UPSTREAM_JOB_NAME = N/A
UPSTREAM_JOB_NUMBER = N/A
CUSTOMIZED_SDK_URL = (leave empty)
```
- To run with OpenJDK9 Hotspot, please use `JAVA_IMPL=openj9` and `JVM_VERSION=openjd9`.
- To run with OpenJDK8-OpenJ9, please use `JAVA_IMPL=openj9` and `JVM_VERSION=openjd8-openj9`.
- To run with OpenJDK9-OpenJ9, please use `JAVA_IMPL=openj9` and `JVM_VERSION=openjd9-openj9`.

**To reproduce manually on a local machine**

One way to reproduce manually 
-  Run the personal build once (this will checkout, compile, and run the test),
-  Ensuring the build doesn't wipe out workspace after completion (this is already done in my personal branch `addJlmTestToAdoptOpenJDK`, so, no action is necessary)
- Log onto the Jenkins slave machine on which the job was run
- CD into the job folder and open the execute.pl script (e.g. /home/jenkins/workspace/test_personal/openjdk-tests/TestConfig/test_output_1522091742698/TestJlmLocal_0/20180326-191543-TestJlmLocal/execute.pl)
 - Copy the Java command that runs the test (inside execute.pl, copy the the command line under "step 1").
 - Run the command with any Java SDK of your choice (e.g. OpenJDK8, OpenJDK8-OpenJ9 etc).

For any other question/queries regarding how to reproduce the issue, please contact the test team.

**Comment by pshipton** [comment]
Can you get the Report output (turn on logging or something) which shows what API the test is calling on OpenJDK when it fails? I see in the ThreadData.writeMBeanInfo(MBeanServerConnection mbsc, ObjectName objectName) code its writing the information to a Report object, but I don't see this information output in the console. 

It appears the test is looping over and calling a number of APIs. I assume there is some incompatibility between OpenJ9 and OpenJDK and the test is hard coded for OpenJ9.

**Comment by Mesbah-Alam** [comment]
I obtained the log produced by the test off of the OpenJDK 8 run, but it doesn't really tell much: 

```
***************************************************************************************************
                                           START TESTING                                           
***************************************************************************************************

****************************************************************************************************
                               VM DATA RETRIEVED USING BEANS DIRECTLY                               
****************************************************************************************************

Information produced at Mar 28, 2018 8:48:19 PM


SYSTEM INFORMATION
  Machine architecture:  amd64
  Operating System:      Linux (4.4.0-116-generic)
  Available processors:  4
VM INFORMATION
  VM version info:       OpenJDK 64-Bit Server VM, Oracle Corporation, 25.71-b00
  VM specification info: Java Virtual Machine Specification, Oracle Corporation, 1.8
  VM id:                 6836@test-softlayer-ubuntu1604-x64-1.adoptopenjdk.net
  Management interface:  1.2

  VM start time:         Mar 28, 2018 8:48:19 PM
  VM uptime:             0.299 seconds

  Program arguments:     


CLASS & LIBRARY PATH INFORMATION
  Classpath:             /home/jenkins/workspace/test_personal/jvmtest/systemtest/op
                         njdk-systemtest/openjdk.test.jlm/bin:/home/jenkins/workspac
                         /test_personal/jvmtest/systemtest/systemtest_prereqs/junit-
                         .12/junit-4.12.jar
  Boot Classpath:        /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-i
                         age/jre/lib/resources.jar:/home/jenkins/workspace/test_pers
                         nal/openjdkbinary/j2sdk-image/jre/lib/rt.jar:/home/jenkins/
                         orkspace/test_personal/openjdkbinary/j2sdk-image/jre/lib/su
                         rsasign.jar:/home/jenkins/workspace/test_personal/openjdkbi
                         ary/j2sdk-image/jre/lib/jsse.jar:/home/jenkins/workspace/te
                         t_personal/openjdkbinary/j2sdk-image/jre/lib/jce.jar:/home/
                         enkins/workspace/test_personal/openjdkbinary/j2sdk-image/jr
                         /lib/charsets.jar:/home/jenkins/workspace/test_personal/ope
                         jdkbinary/j2sdk-image/jre/lib/jfr.jar:/home/jenkins/workspa
                         e/test_personal/openjdkbinary/j2sdk-image/jre/classes
  Library path:          /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib


SYSTEM PROPERTIES
  awt.toolkit = sun.awt.X11.XToolkit
  file.encoding = UTF-8
  file.encoding.pkg = sun.io
  file.separator = /
  java.awt.graphicsenv = sun.awt.X11GraphicsEnvironment
  java.awt.printerjob = sun.print.PSPrinterJob
  java.class.path = /home/jenkins/workspace/test_personal/jvmtest/systemtest/openjdk
  systemtest/openjdk.test.jlm/bin:/home/jenkins/workspace/test_personal/jvmtest/syst
  mtest/systemtest_prereqs/junit-4.12/junit-4.12.jar
  java.class.version = 52.0
  java.endorsed.dirs = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-ima
  e/jre/lib/endorsed
  java.ext.dirs = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image/jr
  /lib/ext:/usr/java/packages/lib/ext
  java.home = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image/jre
  java.io.tmpdir = /tmp
  java.library.path = /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
  java.runtime.name = OpenJDK Runtime Environment
  java.runtime.version = 1.8.0-adoptopenjdk-jenkins_2018_03_08_17_02-b00
  java.specification.name = Java Platform API Specification
  java.specification.vendor = Oracle Corporation
  java.specification.version = 1.8
  java.vendor = Oracle Corporation
  java.vendor.url = http://java.oracle.com/
  java.vendor.url.bug = http://bugreport.sun.com/bugreport/
  java.version = 1.8.0-adoptopenjdk
  java.vm.info = mixed mode
  java.vm.name = OpenJDK 64-Bit Server VM
  java.vm.specification.name = Java Virtual Machine Specification
  java.vm.specification.vendor = Oracle Corporation
  java.vm.specification.version = 1.8
  java.vm.vendor = Oracle Corporation
  java.vm.version = 25.71-b00
  line.separator = 

  os.arch = amd64
  os.name = Linux
  os.version = 4.4.0-116-generic
  path.separator = :
  sun.arch.data.model = 64
  sun.boot.class.path = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-im
  ge/jre/lib/resources.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk
  image/jre/lib/rt.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-ima
  e/jre/lib/sunrsasign.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk
  image/jre/lib/jsse.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-i
  age/jre/lib/jce.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-imag
  /jre/lib/charsets.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-im
  ge/jre/lib/jfr.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image
  jre/classes
  sun.boot.library.path = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-
  mage/jre/lib/amd64
  sun.cpu.endian = little
  sun.cpu.isalist = 
  sun.io.unicode.encoding = UnicodeLittle
  sun.java.command = net.adoptopenjdk.test.jlm.local.VMLogger jlmtestlocal.log
  sun.java.launcher = SUN_STANDARD
  sun.jnu.encoding = UTF-8
  sun.management.compiler = HotSpot 64-Bit Tiered Compilers
  sun.os.patch.level = unknown
  user.country = US
  user.dir = /home/jenkins/workspace/test_personal/openjdk-tests/TestConfig/test_out
  ut_15222700972694/TestJlmLocal_0/20180328-204817-TestJlmLocal/results
  user.home = /home/jenkins
  user.language = en
  user.name = jenkins
  user.timezone = Etc/UTC


LOGGER INFORMATION
  Logger:                javax.management.snmp
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                global
    Parent:              root
    Level:               PARENT LEVEL

  Logger:                javax.management.notification
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.modelmbean
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.timer
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management
    Parent:              root
    Level:               PARENT LEVEL

  Logger:                javax.management.mlet
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.mbeanserver
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.snmp.daemon
    Parent:              javax.management.snmp
    Level:               PARENT LEVEL

  Logger:                javax.management.relation
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.monitor
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.misc
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                root
    Parent:              root
    Level:               INFO



Class Loading and Compilation Information Retrieved at Mar 28, 2018 8:48:19 PM

CLASS LOADING STATS
  Current Loaded Class Count:  1019
  Total Unloaded Class Count:  0
  Total Loaded Class Count:    1019
  Verbose output enabled:      No


Memory Information Retrieved at Mar 28, 2018 8:48:19 PM
Verbose output is disabled for the memory system


MEMORY STATS
  Heap Memory Usage:        init = 132120576(129024K) used = 2713432(2649K) committed = 126877696(123904K) max = 1868038144(1824256K)
  Non-Heap Memory Usage:    init = 2555904(2496K) used = 8263656(8069K) committed = 9633792(9408K) max = -1(-1K)
  Objects Pending Finalization:  0



GARBAGE COLLECTOR INFORMATION
  Garbage Collector:        PS Scavenge
  Total GC occurrences:     1
  Total time spent in GC:   1 milliseconds
  Part of Memory System:    Yes
  Managed Pools:            PS Eden Space
                            PS Survivor Space
                            
  Garbage Collector:        PS MarkSweep
  Total GC occurrences:     1
  Total time spent in GC:   9 milliseconds
  Part of Memory System:    Yes
  Managed Pools:            PS Eden Space
                            PS Survivor Space
                            PS Old Gen
                            


MEMORY MANAGER INFORMATION
  Memory Manager:           CodeCacheManager
  Part of Memory System:    Yes
  Managed Pools:            Code Cache
                            
  Memory Manager:           Metaspace Manager
  Part of Memory System:    Yes
  Managed Pools:            Metaspace
                            Compressed Class Space
                            
  Memory Manager:           PS Scavenge
  Part of Memory System:    Yes
  Managed Pools:            PS Eden Space
                            PS Survivor Space
                            
  Memory Manager:           PS MarkSweep
  Part of Memory System:    Yes
  Managed Pools:            PS Eden Space
                            PS Survivor Space
                            PS Old Gen
                            

MEMORY POOLS INFORMATION
  Memory Pool:              Code Cache
  Memory Type:              Non-heap memory
  Part of Memory System:    Yes
  Current Memory Usage:     init = 2555904(2496K) used = 1593856(1556K) committed = 2555904(2496K) max = 251658240(245760K)
  Peak Memory Usage:        init = 2555904(2496K) used = 1593856(1556K) committed = 2555904(2496K) max = 251658240(245760K)
  Collection Memory Usage:  this VM does not support 
                            the collection memory usage functionality for this pool.
  Pool Managers:            CodeCacheManager
                            
  Usage Threshold:          0 bytes


  Memory Pool:              Metaspace
  Memory Type:              Non-heap memory
  Part of Memory System:    Yes
  Current Memory Usage:     init = 0(0K) used = 6014328(5873K) committed = 6291456(6144K) max = -1(-1K)
  Peak Memory Usage:        init = 0(0K) used = 6014328(5873K) committed = 6291456(6144K) max = -1(-1K)
  Collection Memory Usage:  this VM does not support 
                            the collection memory usage functionality for this pool.
  Pool Managers:            Metaspace Manager
                            
  Usage Threshold:          0 bytes


  Memory Pool:              Compressed Class Space
  Memory Type:              Non-heap memory
  Part of Memory System:    Yes
  Current Memory Usage:     init = 0(0K) used = 675424(659K) committed = 786432(768K) max = 1073741824(1048576K)
  Peak Memory Usage:        init = 0(0K) used = 675424(659K) committed = 786432(768K) max = 1073741824(1048576K)
  Collection Memory Usage:  this VM does not support 
                            the collection memory usage functionality for this pool.
  Pool Managers:            Metaspace Manager
                            
  Usage Threshold:          0 bytes


  Memory Pool:              PS Eden Space
  Memory Type:              Heap memory
  Part of Memory System:    Yes
  Current Memory Usage:     init = 33554432(32768K) used = 2013464(1966K) committed = 33554432(32768K) max = 689963008(673792K)
  Peak Memory Usage:        init = 33554432(32768K) used = 2684448(2621K) committed = 33554432(32768K) max = 689963008(673792K)
  Collection Memory Usage:  init = 33554432(32768K) used = 0(0K) committed = 33554432(32768K) max = 689963008(673792K)
  Pool Managers:            PS MarkSweep
                            PS Scavenge
                            
  Collection Threshold:     0 bytes


  Memory Pool:              PS Survivor Space
  Memory Type:              Heap memory
  Part of Memory System:    Yes
  Current Memory Usage:     init = 5242880(5120K) used = 0(0K) committed = 5242880(5120K) max = 5242880(5120K)
  Peak Memory Usage:        init = 5242880(5120K) used = 770064(752K) committed = 5242880(5120K) max = 5242880(5120K)
  Collection Memory Usage:  init = 5242880(5120K) used = 0(0K) committed = 5242880(5120K) max = 5242880(5120K)
  Pool Managers:            PS MarkSweep
                            PS Scavenge
                            
  Collection Threshold:     0 bytes


  Memory Pool:              PS Old Gen
  Memory Type:              Heap memory
  Part of Memory System:    Yes
  Current Memory Usage:     init = 88080384(86016K) used = 699968(683K) committed = 88080384(86016K) max = 1400897536(1368064K)
  Peak Memory Usage:        init = 88080384(86016K) used = 699968(683K) committed = 88080384(86016K) max = 1400897536(1368064K)
  Collection Memory Usage:  init = 88080384(86016K) used = 699968(683K) committed = 88080384(86016K) max = 1400897536(1368064K)
  Pool Managers:            PS MarkSweep
                            
  Usage Threshold:          0 bytes

  Collection Threshold:     0 bytes




Thread Contention Monitoring is supported and not enabled
Thread Information Retrieved at Mar 28, 2018 8:48:19 PM

THREAD STATS
    Current Thread Count:      4
    Started Thread Count:      4
    Peak Thread Count:         4
    Daemon Thread Count:       3


ALL THREAD INFORMATION
    Object Monitor Usage is supported
    Synchronizer Usage is supported
    ThreadMXBean.dumpAllThreads(boolean,boolean) is supported
        ThreadMXBean.dumpAllThreads(boolean,boolean) and ThreadMXBean.getThreadInfo(ThreadMXBean.getAllThreads(long[], int)
 return the same  number of ThreadInfo objects 
        ThreadMXBean.dumpAllThreads(boolean,boolean) and ThreadMXBean.getThreadInfo(ThreadMXBean.getAllThreads(long[], boolean, boolean)
 return the same  number of ThreadInfo objects 
        Using information from ThreadMXBean.dumpAllThreads()

    Thread CPU Time is supported and enabled

  *****************************
  Thread: Signal Dispatcher (4)
  *****************************

            Thread State:              RUNNABLE
            Thread Suspended:          No
            Thread in native code:     No

            # monitor blocks occurred: 0
            # waits for notification:  0
            Currently blocked by:      Thread not currently blocked

        
        CPU Time:                  0.00 seconds
        User Time:                 -1 seconds


  *********************
  Thread: Finalizer (3)
  *********************

        Thread State:              WAITING
        Thread Suspended:          No
        Thread in native code:     No

        # monitor blocks occurred: 2
        # waits for notification:  3
        Currently blocked by:      java.lang.ref.ReferenceQueue$Lock@723279cf

        Stack trace:               java.lang.Object.wait(Native Method)
                                   java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
                                   java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
                                   java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:216)
                                   
    
    CPU Time:                  0.00 seconds
    User Time:                 -1 seconds


  *****************************
  Thread: Reference Handler (2)
  *****************************

    Thread State:              WAITING
    Thread Suspended:          No
    Thread in native code:     No

    # monitor blocks occurred: 2
    # waits for notification:  2
    Currently blocked by:      java.lang.ref.Reference$Lock@10f87f48

    Stack trace:               java.lang.Object.wait(Native Method)
                               java.lang.Object.wait(Object.java:502)
                               java.lang.ref.Reference.tryHandlePending(Reference.java:191)
                               java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)
                               

    CPU Time:                  0.00 seconds
    User Time:                 -1 seconds


  ****************
  Thread: main (1)
  ****************

    Thread State:              RUNNABLE
    Thread Suspended:          No
    Thread in native code:     No

    # monitor blocks occurred: 0
    # waits for notification:  0
    Currently blocked by:      Thread not currently blocked

    Stack trace:               sun.management.ThreadImpl.getThreadInfo1(Native Method)
                               sun.management.ThreadImpl.getThreadInfo(ThreadImpl.java:178)
                               net.adoptopenjdk.test.jlm.resources.ThreadData.writeData(ThreadData.java:150)
                               net.adoptopenjdk.test.jlm.local.VMLogger.writeProxyData(VMLogger.java:222)
                               net.adoptopenjdk.test.jlm.local.VMLogger.main(VMLogger.java:127)
                               

    CPU Time:                  0.30 seconds
    User Time:                 -1 seconds



  *****************************
  Thread: Signal Dispatcher (4)
  *****************************

        Thread State:              RUNNABLE
        Thread Suspended:          No
        Thread in native code:     No

        # monitor blocks occurred: 0
        # waits for notification:  0
        Currently blocked by:      Thread not currently blocked

  ****************
  Lock Information
  ****************
            Thread is not locked
            Locked synchronizers:      none
            Monitor locks:             none
        
    CPU Time:                  0.00 seconds
    User Time:                 0.00 seconds


  *********************
  Thread: Finalizer (3)
  *********************

        Thread State:              WAITING
        Thread Suspended:          No
        Thread in native code:     No

        # monitor blocks occurred: 2
        # waits for notification:  3
        Currently blocked by:      java.lang.ref.ReferenceQueue$Lock@723279cf

        Stack trace:               java.lang.Object.wait(Native Method)
                                   java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
                                   java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
                                   java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:216)

  ****************
  Lock Information
  ****************
            Thread is locked:          java.lang.ref.ReferenceQueue$Lock@723279cf
                Class name: "java.lang.ref.ReferenceQueue$Lock"; Identity hash code: "1915910607"

            Locked synchronizers:      none
            Monitor locks:             none
        
    CPU Time:                  0.00 seconds
    User Time:                 0.00 seconds


  *****************************
  Thread: Reference Handler (2)
  *****************************

        Thread State:              WAITING
        Thread Suspended:          No
        Thread in native code:     No

        # monitor blocks occurred: 2
        # waits for notification:  2
        Currently blocked by:      java.lang.ref.Reference$Lock@10f87f48

        Stack trace:               java.lang.Object.wait(Native Method)
                                   java.lang.Object.wait(Object.java:502)
                                   java.lang.ref.Reference.tryHandlePending(Reference.java:191)
                                   java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

  ****************
  Lock Information
  ****************
            Thread is locked:          java.lang.ref.Reference$Lock@10f87f48
                Class name: "java.lang.ref.Reference$Lock"; Identity hash code: "284720968"

            Locked synchronizers:      none
            Monitor locks:             none
        
    CPU Time:                  0.00 seconds
    User Time:                 0.00 seconds


  ****************
  Thread: main (1)
  ****************

        Thread State:              RUNNABLE
        Thread Suspended:          No
        Thread in native code:     No

        # monitor blocks occurred: 0
        # waits for notification:  0
        Currently blocked by:      Thread not currently blocked

        Stack trace:               sun.management.ThreadImpl.dumpThreads0(Native Method)
                                   sun.management.ThreadImpl.getThreadInfo(ThreadImpl.java:448)
                                   net.adoptopenjdk.test.jlm.resources.ThreadData.writeData(ThreadData.java:258)
                                   net.adoptopenjdk.test.jlm.local.VMLogger.writeProxyData(VMLogger.java:222)
                                   net.adoptopenjdk.test.jlm.local.VMLogger.main(VMLogger.java:127)

  ****************
  Lock Information
  ****************
            Thread is not locked
            Locked synchronizers:      none
            Monitor locks:             none
        
    CPU Time:                  0.30 seconds
    User Time:                 0.22 seconds



MONITOR DEADLOCK THREAD INFORMATION

    ThreadMXBean.findMonitorDeadlockedThreads() returned null

DEADLOCK THREAD INFORMATION

    Currently no deadlocked threads.

************  NO FAILURES RECORDED  ************
****************************************************************************************************
                              VM DATA RETRIEVED USING THE MBEAN SERVER                              
****************************************************************************************************

Information produced at Mar 28, 2018 8:48:19 PM


SYSTEM INFORMATION
  Machine architecture:  amd64
  Operating System:      Linux (4.4.0-116-generic)
  Available processors:  4
  System load average:   2.1
VM INFORMATION
  VM version info:       OpenJDK 64-Bit Server VM, Oracle Corporation, 25.71-b00
  VM specification info: Java Virtual Machine Specification, Oracle Corporation, 1.8
  VM id:                 6836@test-softlayer-ubuntu1604-x64-1.adoptopenjdk.net
  Management interface:  1.2

  VM start time:         Mar 28, 2018 8:48:19 PM
  VM uptime:             0.323 seconds

  Program arguments:     


CLASS & LIBRARY PATH INFORMATION
  Classpath:             /home/jenkins/workspace/test_personal/jvmtest/systemtest/op
                         njdk-systemtest/openjdk.test.jlm/bin:/home/jenkins/workspac
                         /test_personal/jvmtest/systemtest/systemtest_prereqs/junit-
                         .12/junit-4.12.jar
  Boot Classpath:        /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-i
                         age/jre/lib/resources.jar:/home/jenkins/workspace/test_pers
                         nal/openjdkbinary/j2sdk-image/jre/lib/rt.jar:/home/jenkins/
                         orkspace/test_personal/openjdkbinary/j2sdk-image/jre/lib/su
                         rsasign.jar:/home/jenkins/workspace/test_personal/openjdkbi
                         ary/j2sdk-image/jre/lib/jsse.jar:/home/jenkins/workspace/te
                         t_personal/openjdkbinary/j2sdk-image/jre/lib/jce.jar:/home/
                         enkins/workspace/test_personal/openjdkbinary/j2sdk-image/jr
                         /lib/charsets.jar:/home/jenkins/workspace/test_personal/ope
                         jdkbinary/j2sdk-image/jre/lib/jfr.jar:/home/jenkins/workspa
                         e/test_personal/openjdkbinary/j2sdk-image/jre/classes
  Library path:          /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib


SYSTEM PROPERTIES
  awt.toolkit = sun.awt.X11.XToolkit
  file.encoding = UTF-8
  file.encoding.pkg = sun.io
  file.separator = /
  java.awt.graphicsenv = sun.awt.X11GraphicsEnvironment
  java.awt.printerjob = sun.print.PSPrinterJob
  java.class.path = /home/jenkins/workspace/test_personal/jvmtest/systemtest/openjdk
  systemtest/openjdk.test.jlm/bin:/home/jenkins/workspace/test_personal/jvmtest/syst
  mtest/systemtest_prereqs/junit-4.12/junit-4.12.jar
  java.class.version = 52.0
  java.endorsed.dirs = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-ima
  e/jre/lib/endorsed
  java.ext.dirs = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image/jr
  /lib/ext:/usr/java/packages/lib/ext
  java.home = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image/jre
  java.io.tmpdir = /tmp
  java.library.path = /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
  java.runtime.name = OpenJDK Runtime Environment
  java.runtime.version = 1.8.0-adoptopenjdk-jenkins_2018_03_08_17_02-b00
  java.specification.name = Java Platform API Specification
  java.specification.vendor = Oracle Corporation
  java.specification.version = 1.8
  java.vendor = Oracle Corporation
  java.vendor.url = http://java.oracle.com/
  java.vendor.url.bug = http://bugreport.sun.com/bugreport/
  java.version = 1.8.0-adoptopenjdk
  java.vm.info = mixed mode
  java.vm.name = OpenJDK 64-Bit Server VM
  java.vm.specification.name = Java Virtual Machine Specification
  java.vm.specification.vendor = Oracle Corporation
  java.vm.specification.version = 1.8
  java.vm.vendor = Oracle Corporation
  java.vm.version = 25.71-b00
  line.separator = 

  os.arch = amd64
  os.name = Linux
  os.version = 4.4.0-116-generic
  path.separator = :
  sun.arch.data.model = 64
  sun.boot.class.path = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-im
  ge/jre/lib/resources.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk
  image/jre/lib/rt.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-ima
  e/jre/lib/sunrsasign.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk
  image/jre/lib/jsse.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-i
  age/jre/lib/jce.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-imag
  /jre/lib/charsets.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-im
  ge/jre/lib/jfr.jar:/home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-image
  jre/classes
  sun.boot.library.path = /home/jenkins/workspace/test_personal/openjdkbinary/j2sdk-
  mage/jre/lib/amd64
  sun.cpu.endian = little
  sun.cpu.isalist = 
  sun.io.unicode.encoding = UnicodeLittle
  sun.java.command = net.adoptopenjdk.test.jlm.local.VMLogger jlmtestlocal.log
  sun.java.launcher = SUN_STANDARD
  sun.jnu.encoding = UTF-8
  sun.management.compiler = HotSpot 64-Bit Tiered Compilers
  sun.os.patch.level = unknown
  user.country = US
  user.dir = /home/jenkins/workspace/test_personal/openjdk-tests/TestConfig/test_out
  ut_15222700972694/TestJlmLocal_0/20180328-204817-TestJlmLocal/results
  user.home = /home/jenkins
  user.language = en
  user.name = jenkins
  user.timezone = Etc/UTC


LOGGER INFORMATION
  Logger:                javax.management.snmp
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                global
    Parent:              root
    Level:               PARENT LEVEL

  Logger:                javax.management.notification
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.modelmbean
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.timer
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management
    Parent:              root
    Level:               PARENT LEVEL

  Logger:                javax.management.mlet
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.mbeanserver
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.snmp.daemon
    Parent:              javax.management.snmp
    Level:               PARENT LEVEL

  Logger:                javax.management.relation
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.monitor
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                javax.management.misc
    Parent:              javax.management
    Level:               PARENT LEVEL

  Logger:                root
    Parent:              root
    Level:               INFO



Class Loading and Compilation Information Retrieved at Mar 28, 2018 8:48:19 PM

CLASS LOADING STATS
  Current Loaded Class Count:  1053
  Total Unloaded Class Count:  0
  Total Loaded Class Count:    1053
  Verbose output enabled:      No


** System property java.compiler information NOT available **
```

**Comment by Mesbah-Alam** [comment]
I am attaching here the logs produced by runs 2 runs using OpenJDK8 and OpenJDK8-OpenJ9 below: 
[openjdk8_jlmtestlocal.log](https://github.com/eclipse/openj9/files/1857786/openjdk8_jlmtestlocal.log)
[openjdk8-openj9_jlmtestlocal.log](https://github.com/eclipse/openj9/files/1857787/openjdk8-openj9_jlmtestlocal.log)

The logs don't say much about which class is being invoked by the test though!

**Comment by pshipton** [comment]
Maybe not the right logs? Look at the code in openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java above line 866. Its using the Report class to write information about what is being called. 

Comparing the logs, it seems to be going wrong obtaining the data for MemoryData.writeData(), but the Report data would be very helpful.

**Comment by Mesbah-Alam** [comment]
The `Report` class is initialized with a PrintWriter constructed with the given log file (attached above) supplied in args[0] of VMLogger-- the main class of this test. Hence, methods such as `Report.printlnIndent()` are in fact writing to this log file (e.g., in ThreadData.java, line 415, `Report.printlnIndent("Current Thread Count:..)` is printing the output in the log file).

**Comment by pshipton** [comment]
Then the log file hasn't been flushed are we aren't getting the most recent updates.

**Comment by pshipton** [comment]
Its a bit weird because ThreadData.java:1093 does "Report.flushAll();".

**Comment by pshipton** [comment]
When you look at the executing code shown by the stack traces, you can see output is missing from the log. I don't see anything wrong with the logging code, and there is a flush(), I'm not sure why the output is missing. @Mesbah-Alam is going to try running under the Java debugger to track down the problem, which appears to be a incompatibility between the test and OpenJDK.

**Comment by Mesbah-Alam** [comment]
The issue is also seen in Openjdk10-Openj9 (the only Openj9 version in which the error is seen) : https://ci.adoptopenjdk.net/view/System%20tests/job/openjdk10_j9_systemtest_x86-64_linux/7/tapResults/

**Comment by lumpfish** [comment]
This is caused by overloaded mxbean operations (methods) being called with incorrect arguments.

The test first uses reflection to find the mxbean methods and generates arguments for them, then uses reflection again to find the method and execute it.  However, when the method is found the second time no check is made to see if it has the correct signature for the arguments which have been generated, so for overloaded methods there is a random chance that the incorrect method will be selected.

Working on a fix to loop though the available mxbean operations until the one with the correct signature is found.

**Comment by lumpfish** [comment]
PR to fix is https://github.com/AdoptOpenJDK/openjdk-systemtest/pull/364.

The fix exposed a separate issue in openj9: https://github.com/eclipse/openj9/issues/10796.  PR https://github.com/AdoptOpenJDK/openjdk-systemtest/pull/364 includes a check to avoid hitting https://github.com/eclipse/openj9/issues/10796 pending an openj9 fix.

The tests are currently excluded via the systemtest playlist.xml file, so merging this PR will not cause the tests to start executing.  PR https://github.com/AdoptOpenJDK/openjdk-tests/pull/2003 updates the playlist.xml file to reenable the tests.

Test runs (which were run with both PRs) with both PRs are:
jdk8

openj9

x86-64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4042
ppc64_aix: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4043
ppc64le_linux_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4044
s390x_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4045
x86-64_mac: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4046
x86-64_windows_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4047
x86-32_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4048

hotspot

x86-64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4049
ppc64_aix: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4050
ppc64le_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4056
s390x_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4052
x86-64_mac: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4053
x86-64_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4054
x86-32_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4055


jdk11

openj9

x86-64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4085
ppc64_aix: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4088
ppc64le_linux_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4089
s390x_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4091
x86-64_mac: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4092
x86-64_windows_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4093
aarch64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4094

hotspot

x86-64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4096
ppc64_aix: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4100
ppc64le_linux_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4101
s390x_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4102
x86-64_mac: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4097
x86-64_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4098
x86-32_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4099
aarch64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4103
aarch32_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4108


jdk15

openj9

x86-64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4109
ppc64_aix: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4110
ppc64le_linux_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4111
s390x_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4112
x86-64_mac: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4113
x86-64_windows_xl: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4128

hotspot

x86-64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4118
ppc64_aix: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4122
ppc64le_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4123
s390x_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4124
x86-64_mac: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4119
x86-64_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4129
x86-32_windows: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4130
aarch64_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4125
aarch32_linux: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4127

**Comment by Mesbah-Alam** [comment]
Hi Simon, Thanks for cleaning up the test bug. 

> PR AdoptOpenJDK/openjdk-systemtest#364 includes a check to avoid hitting #10796 pending an openj9 fix.

Do we need to remove this check from the test once https://github.com/eclipse/openj9/issues/10796 is resolved? 

@lumpfish

**Comment by Mesbah-Alam** [comment]
Please ignore my question above. Your [comment](https://github.com/AdoptOpenJDK/openjdk-systemtest/blob/e87fff6998bfcb77c59b893ab8b31adde6ddc820/openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java#L1084) in the PR answers it.
